client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/protocol/Codec40.java

```diff
@@ -0,0 +1,53 @@
+package org.infinispan.client.hotrod.impl.protocol;
+
+import java.util.Map;
+
+import org.infinispan.client.hotrod.DataFormat;
+import org.infinispan.client.hotrod.MetadataValue;
+import org.infinispan.client.hotrod.impl.operations.GetWithMetadataOperation;
+import org.infinispan.client.hotrod.impl.transport.netty.ByteBufUtil;
+import org.infinispan.commons.configuration.ClassAllowList;
+import org.infinispan.commons.marshall.Marshaller;
+
+import io.netty.buffer.ByteBuf;
+
+/**
+ * @since 14.0
+ */
+public class Codec40 extends Codec31 {
+
+   @Override
+   public Object returnPossiblePrevValue(ByteBuf buf, short status, DataFormat dataFormat, int flags, ClassAllowList allowList, Marshaller marshaller) {
+      if (HotRodConstants.hasPrevious(status)) {
+         MetadataValue<Object> metadataValue = GetWithMetadataOperation.readMetadataValue(buf, status, dataFormat, allowList);
+         return metadataValue != null ? metadataValue.getValue() : null;
+      } else {
+         return null;
+      }
+   }
+
+   @Override
+   public HeaderParams writeHeader(ByteBuf buf, HeaderParams params) {
+      return writeHeader(buf, params, HotRodConstants.VERSION_40);
+   }
+
+   @Override
+   protected HeaderParams writeHeader(ByteBuf buf, HeaderParams params, byte version) {
+      HeaderParams headerParams = super.writeHeader(buf, params, version);
+      writeOtherParams(buf, params.otherParams());
+      return headerParams;
+   }
+
+   private void writeOtherParams(ByteBuf buf, Map<String, String> parameters) {
```

Comment:
I commented on Fabio's PR... I think we decided to make the value a `byte[]`

Reply:
Sure, I would say any changes in there @fax4ever can let me know and I can squash any of his review changes